### ISSUE TITLE
Exercise build error for Android: Failed to create JavaTypeInfo for class: IAccessibilityStateChangeListenerImplemento

### ISSUE THREAD

**Issue by randroid88** [MicrosoftDocs/mslearn-customize-layout-in-xamarin-forms-xaml-pages#1]
These projects can't be run on Android in the latest Visual Studio 2019.

When I try to build I get this error:
`Severity	Code	Description	Project	File	Line	Suppression State
Error		Failed to create JavaTypeInfo for class: Android.Support.V4.View.Accessibility.AccessibilityManagerCompat/IAccessibilityStateChangeListenerImplementor due to System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\Development\source\repo\mslearn-xamarin-forms-customize-xaml-layout-master\src\exercise2\start\TipCalculator.Android\obj\Debug\81\android\src\mono\android\support\v4\view\accessibility\AccessibilityManagerCompat_AccessibilityStateChangeListenerImplementor.java'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at Xamarin.Android.Tools.Files.CopyIfStreamChanged(Stream stream, String destination)
   at Xamarin.Android.Tasks.Generator.CreateJavaSources(TaskLoggingHelper log, IEnumerable`1 javaTypes, String outputPath, String applicationJavaClass, Boolean useSharedRuntime, Boolean generateOnCreateOverrides, Boolean hasExportReference)	TipCalculator.Android			
`

The work around is to create a new Xamarin Forms project and copy over the code.

Info:

> Microsoft Visual Studio Community 2019
> Version 16.1.1
> VisualStudio.16.Release/16.1.1+28922.388
> Microsoft .NET Framework
> Version 4.7.03056
> 
> Installed Version: Community
> 
> Azure App Service Tools v3.0.0   16.1.429.50124
> Azure App Service Tools v3.0.0
> 
> C# Tools   3.1.0-beta4-19266-03+9d80dea7fe1b14043b9b2ac4d0b59ed26f508742
> C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.
> 
> Common Azure Tools   1.10
> Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.
> 
> Extensibility Message Bus   1.1.77 (master@24013d5)
> Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.
> 
> IntelliCode Extension   1.0
> IntelliCode Visual Studio Extension Detailed Info
> 
> JetBrains ReSharper Ultimate 2019.1.1   Build 191.0.20190501.122851
> JetBrains ReSharper Ultimate package for Microsoft Visual Studio. For more information about ReSharper Ultimate, visit http://www.jetbrains.com/resharper. Copyright © 2019 JetBrains, Inc.
> 
> Microsoft JVM Debugger   1.0
> Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines
> 
> Microsoft MI-Based Debugger   1.0
> Provides support for connecting Visual Studio to MI compatible debuggers
> 
> Mono Debugging for Visual Studio   16.1.1 (2473f22)
> Support for debugging Mono processes with Visual Studio.
> 
> NuGet Package Manager   5.1.0
> NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/
> 
> ProjectServicesPackage Extension   1.0
> ProjectServicesPackage Visual Studio Extension Detailed Info
> 
> ResourcePackage Extension   1.0
> ResourcePackage Visual Studio Extension Detailed Info
> 
> ResourcePackage Extension   1.0
> ResourcePackage Visual Studio Extension Detailed Info
> 
> Visual Basic Tools   3.1.0-beta4-19266-03+9d80dea7fe1b14043b9b2ac4d0b59ed26f508742
> Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.
> 
> Visual F# Tools 10.4 for F# 4.6   16.1.0-beta.19253.3+42526fe359672a05fd562dc16a91a43d0fe047a7
> Microsoft Visual F# Tools 10.4 for F# 4.6
> 
> Visual Studio Code Debug Adapter Host Package   1.0
> Interop layer for hosting Visual Studio Code debug adapters in Visual Studio
> 
> VisualStudio.Mac   1.0
> Mac Extension for Visual Studio
> 
> Xamarin   16.1.0.542 (d16-1@68b985244)
> Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.
> 
> Xamarin Designer   16.1.0.418 (remotes/origin/d16-1@5b958bb10)
> Visual Studio extension to enable Xamarin Designer tools in Visual Studio.
> 
> Xamarin Templates   16.2.112 (4db4af4)
> Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.
> 
> Xamarin.Android SDK   9.3.0.22 (HEAD/8e7764fdf)
> Xamarin.Android Reference Assemblies and MSBuild support.
>     Mono: mono/mono/2018-08@3cb36842fc4
>     Java.Interop: xamarin/java.interop/d16-1@5ddc3e3
>     LibZipSharp: grendello/LibZipSharp/d16-1@44de300
>     LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
>     ProGuard: xamarin/proguard/master@905836d
>     SQLite: xamarin/sqlite/3.27.1@8212a2d
>     Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-1@acabd26
> 
> 
> Xamarin.iOS and Xamarin.Mac SDK   12.10.0.153 (750a879)
> Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Comment by randroid88** [comment]
Looks like a max file path length problem: https://developercommunity.visualstudio.com/content/problem/521034/failed-to-create-javatypeinfo.html
